### PR TITLE
Upgrade to CesiumJS v1.92

### PR DESCRIPTION
**Pull request by kring**
This is a PR into #6211 for ease of review.

Requires TerriaJS/cesium#84.

This looks like a lot of changes, but most of them are just to remove `makeRealPromise`, because CesiumJS uses real promises now.

### CODE REVIEW

**Comment by AnaBelgun** [comment]
Tested almost all layers on TerriaMap / ci and no issues found. 
Test these examples too: 

- [x] 3d buildings - e.g https://qld.digitaltwin.terria.io/api/v0/data/2eeeadde-42a9-4372-aaec-abba6275a504/NPB/tileset.json + styling testing file [here](https://csiroau.sharepoint.com/sites/WebDataPlatformsgroup/_layouts/15/download.aspx?UniqueId=9e3ae8f9d16a4348aeca771ce4c47bff&e=BxSe1B)
- [x] reality meshes - e.g https://qld.digitaltwin.terria.io/api/v0/data/5f0c5ad9-a91d-4bae-88e1-d78553743c8e/Brisbane_75mm_UM_3dtiles/tileset.json 

WFS and Esri feature servers - e.g. 

- [x] https://portal.spatial.nsw.gov.au/server/rest/services/SurveyMarkGDA94/FeatureServer  
- [x] http://services.ga.gov.au/gis/critical-minerals/wfs

File: lib/ModelMixins/TileErrorHandlerMixin.ts [hunk 1]
```diff
@@ -194,7 +200,7 @@ function TileErrorHandlerMixin<T extends Constructor<ModelType>>(Base: T) {
       // By setting retry to a promise, we tell cesium/leaflet to
       // reload the tile if the promise resolves successfully
       // https://github.com/TerriaJS/cesium/blob/terriajs/Source/Core/TileProviderError.js#L161
-      tileProviderError.retry = result;
+      tileProviderError.retry = promise;
```
Comment (nf-s) [inline lib/ModelMixins/TileErrorHandlerMixin.ts:204]: image

File: lib/ModelMixins/TileErrorHandlerMixin.ts [hunk 2]
```diff
@@ -194,7 +200,7 @@ function TileErrorHandlerMixin<T extends Constructor<ModelType>>(Base: T) {
...
```
Comment (nf-s) [inline lib/ModelMixins/TileErrorHandlerMixin.ts:203]: `ImageryLayer.js` in TerriaJS/cesium is still using `when`

So when triggering the `TileErrorHandlerMixin` we are getting these errors

image

Comment (nf-s) [reply lib/ModelMixins/TileErrorHandlerMixin.ts:203]: [Test link](http://ci.terria.io/update-to-cesium-192/#start=%7B%22version%22%3A%228.0.0%22%2C%22initSources%22%3A%5B%7B%22stratum%22%3A%22user%22%2C%22models%22%3A%7B%22__User-Added_Data__%22%3A%7B%22isOpen%22%3Atrue%2C%22members%22%3A%5B%22%2Fhttps%3A%2F%2Fportal.spatial.nsw.gov.au%2Fserver%2Frest%2Fservices%2FSurveyMarkGDA94%2FFeatureServer%22%2C%22%2Fhttp%3A%2F%2Fservices.ga.gov.au%2Fgis%2Fcritical-minerals%2Fwfs%22%2C%22699843ef-7a35-45d2-a1ee-3d6cdeb1eed0%22%2C%22%2Fhttps%3A%2F%2Fqld.digitaltwin.terria.io%2Fapi%2Fv0%2Fdata%2F5f0c5ad9-a91d-4bae-88e1-d78553743c8e%2FBrisbane_75mm_UM_3dtiles%2Ftileset.json%22%5D%2C%22knownContainerUniqueIds%22%3A%5B%22%2F%22%5D%2C%22type%22%3A%22group%22%7D%2C%22FxPHcHvxoO%22%3A%7B%22isOpen%22%3Atrue%2C%22knownContainerUniqueIds%22%3A%5B%22%2F%22%5D%2C%22type%22%3A%22wms-group%22%7D%2C%22%2F%2FSurface+Geology%22%3A%7B%22splitDirection%22%3A1%2C%22knownContainerUniqueIds%22%3A%5B%22%2F%22%5D%2C%22type%22%3A%22esri-mapServer%22%7D%2C%22%2Fhttps%3A%2F%2Fportal.spatial.nsw.gov.au%2Fserver%2Frest%2Fservices%2FSurveyMarkGDA94%2FFeatureServer%22%3A%7B%22url%22%3A%22https%3A%2F%2Fportal.spatial.nsw.gov.au%2Fserver%2Frest%2Fservices%2FSurveyMarkGDA94%2FFeatureServer%22%2C%22dereferenced%22%3A%7B%22isOpen%22%3Atrue%2C%22url%22%3A%22https%3A%2F%2Fportal.spatial.nsw.gov.au%2Fserver%2Frest%2Fservices%2FSurveyMarkGDA94%2FFeatureServer%22%7D%2C%22knownContainerUniqueIds%22%3A%5B%22__User-Added_Data__%22%5D%2C%22type%22%3A%22url-reference%22%7D%2C%22%2Fhttps%3A%2F%2Fportal.spatial.nsw.gov.au%2Fserver%2Frest%2Fservices%2FSurveyMarkGDA94%2FFeatureServer%2F0%22%3A%7B%22styles%22%3A%5B%7B%22id%22%3A%22ESRI%22%2C%22point%22%3A%7B%22enum%22%3A%5B%7B%22marker%22%3A%22mountain%22%2C%22height%22%3A30%2C%22width%22%3A40%7D%5D%7D%7D%5D%2C%22knownContainerUniqueIds%22%3A%5B%22%2Fhttps%3A%2F%2Fportal.spatial.nsw.gov.au%2Fserver%2Frest%2Fservices%2FSurveyMarkGDA94%2FFeatureServer%22%5D%2C%22type%22%3A%22esri-featureServer%22%7D%2C%22%2Fhttp%3A%2F%2Fservices.ga.gov.au%2Fgis%2Fcritical-minerals%2Fwfs%22%3A%7B%22isOpen%22%3Atrue%2C%22url%22%3A%22http%3A%2F%2Fservices.ga.gov.au%2Fgis%2Fcritical-minerals%2Fwfs%22%2C%22knownContainerUniqueIds%22%3A%5B%22__User-Added_Data__%22%5D%2C%22type%22%3A%22wfs-group%22%7D%2C%22%2Fhttp%3A%2F%2Fservices.ga.gov.au%2Fgis%2Fcritical-minerals%2Fwfs%2Fcmmi%3ACD_prospectivity_model%22%3A%7B%22styles%22%3A%5B%7B%22id%22%3A%22Period_Minimum%22%2C%22color%22%3A%7B%22colorPalette%22%3A%22HighContrast%22%7D%2C%22outline%22%3A%7B%22null%22%3A%7B%22color%22%3A%22rgba%280%2C0%2C0%2C0%29%22%2C%22width%22%3A0%7D%7D%2C%22hidden%22%3Afalse%7D%5D%2C%22activeStyle%22%3A%22Period_Minimum%22%2C%22knownContainerUniqueIds%22%3A%5B%22%2Fhttp%3A%2F%2Fservices.ga.gov.au%2Fgis%2Fcritical-minerals%2Fwfs%22%5D%2C%22type%22%3A%22wfs%22%7D%2C%22%2Fhttp%3A%2F%2Fservices.ga.gov.au%2Fgis%2Fcritical-minerals%2Fwfs%2Fcmmi%3ACriticalMineralDepositsGeochemistry%22%3A%7B%22show%22%3Atrue%2C%22activeStyle%22%3A%22AL2O3_WT_PERCENT%22%2C%22knownContainerUniqueIds%22%3A%5B%22%2Fhttp%3A%2F%2Fservices.ga.gov.au%2Fgis%2Fcritical-minerals%2Fwfs%22%5D%2C%22type%22%3A%22wfs%22%7D%2C%22%2Fhttp%3A%2F%2Fservices.ga.gov.au%2Fgis%2Fcritical-minerals%2Fwfs%2Fcmmi%3AMVT_prospectivity_model%22%3A%7B%22show%22%3Atrue%2C%22activeStyle%22%3A%22Period_Maximum%22%2C%22knownContainerUniqueIds%22%3A%5B%22%2Fhttp%3A%2F%2Fservices.ga.gov.au%2Fgis%2Fcritical-minerals%2Fwfs%22%5D%2C%22type%22%3A%22wfs%22%7D%2C%22699843ef-7a35-45d2-a1ee-3d6cdeb1eed0%22%3A%7B%22splitSourceItemId%22%3A%22%2F%2FSurface+Geology%22%2C%22dereferenced%22%3A%7B%22opacity%22%3A0%2C%22splitDirection%22%3A-1%2C%22name%22%3A%22Surface+Geology+%28copy%29%22%7D%2C%22knownContainerUniqueIds%22%3A%5B%22__User-Added_Data__%22%5D%2C%22type%22%3A%22split-reference%22%7D%2C%22%2Fhttps%3A%2F%2Fqld.digitaltwin.terria.io%2Fapi%2Fv0%2Fdata%2F5f0c5ad9-a91d-4bae-88e1-d78553743c8e%2FBrisbane_75mm_UM_3dtiles%2Ftileset.json%22%3A%7B%22url%22%3A%22https%3A%2F%2Fqld.digitaltwin.terria.io%2Fapi%2Fv0%2Fdata%2F5f0c5ad9-a91d-4bae-88e1-d78553743c8e%2FBrisbane_75mm_UM_3dtiles%2Ftileset.json%22%2C%22knownContainerUniqueIds%22%3A%5B%22__User-Added_Data__%22%5D%2C%22type%22%3A%223d-tiles%22%7D%2C%22FxPHcHvxoO%2FChl_MIM_n_cloud%22%3A%7B%22currentTime%22%3A%222020-05-01T00%3A00%3A00Z%22%2C%22startTime%22%3A%222002-07-01T00%3A00%3A00Z%22%2C%22stopTime%22%3A%222020-05-01T00%3A00%3A00Z%22%2C%22multiplier%22%3A1308859.546511628%2C%22isPaused%22%3Atrue%2C%22styles%22%3A%22waht%22%2C%22knownContainerUniqueIds%22%3A%5B%22FxPHcHvxoO%22%5D%2C%22type%22%3A%22wms%22%7D%2C%22%2F%22%3A%7B%22type%22%3A%22group%22%7D%7D%2C%22workbench%22%3A%5B%22FxPHcHvxoO%2FChl_MIM_n_cloud%22%5D%2C%22timeline%22%3A%5B%22FxPHcHvxoO%2FChl_MIM_n_cloud%22%5D%2C%22initialCamera%22%3A%7B%22west%22%3A149.92301947457958%2C%22south%22%3A-23.73646205595301%2C%22east%22%3A151.4181892869659%2C%22north%22%3A-22.77753458636704%2C%22position%22%3A%7B%22x%22%3A-5241025.418951651%2C%22y%22%3A2860674.0424447437%2C%22z%22%3A-2525209.673106327%7D%2C%22direction%22%3A%7B%22x%22%3A0.9822386725646133%2C%22y%22%3A0.0842612727180087%2C%22z%22%3A0.167652104187355%7D%2C%22up%22%3A%7B%22x%22%3A-0.0007258885766150414%2C%22y%22%3A0.8951971923144721%2C%22z%22%3A-0.44566967583408734%7D%7D%2C%22homeCamera%22%3A%7B%22west%22%3A109%2C%22south%22%3A-45%2C%22east%22%3A158%2C%22north%22%3A-8%7D%2C%22viewerMode%22%3A%223dSmooth%22%2C%22showSplitter%22%3Afalse%2C%22splitPosition%22%3A0.12656719883889683%2C%22settings%22%3A%7B%22baseMaximumScreenSpaceError%22%3A2%2C%22useNativeResolution%22%3Afalse%2C%22alwaysShowTimeline%22%3Afalse%2C%22baseMapId%22%3A%22basemap-darkmatter%22%2C%22terrainSplitDirection%22%3A0%2C%22depthTestAgainstTerrainEnabled%22%3Afalse%7D%2C%22stories%22%3A%5B%5D%7D%5D%7D)

See console errors

**Review by nf-s — APPROVED** [review]

File: lib/ModelMixins/TileErrorHandlerMixin.ts (hunk 2 above)
Comment (nf-s) [reply lib/ModelMixins/TileErrorHandlerMixin.ts:203]: Fixed in https://github.com/TerriaJS/cesium/pull/85
Bumped terriajs-cesium version to `1.92.0-tile-error-provider-fix-2`

**Pull request merged**